sdk/go/pulumi/context.go

```diff
@@ -1032,12 +1033,24 @@ func (ctx *Context) getOpts(t string, providers map[string]ProviderResource, opt
 	if provider != nil {
 		pr, err := ctx.resolveProviderReference(provider)
 		if err != nil {
-			return "", nil, false, "", false, "", nil, nil, "", err
+			return "", nil, false, "", nil, false, "", nil, nil, "", err
 		}
 		providerRef = pr
 	}
 
-	return parentURN, depURNs, opts.Protect, providerRef, opts.DeleteBeforeReplace,
+	var providerRefs map[string]string
+	if opts.Providers != nil {
```

Comment:
Probably want to make this conditional on whether or not we're preparing inputs for a remote resource.